src/osd/ECMsgTypes.cc

```diff
@@ -220,7 +220,7 @@ void ECSubRead::decode(bufferlist::const_iterator &bl)
     decode(to_read, bl);
   }
   decode(attrs_to_read, bl);
-  if (struct_v > 2 && struct_v > struct_compat) {
+  if (struct_v > 2 && struct_v.v > struct_compat) {
```

Comment:
please explain why you're adding the compare against struct_compat and v ? there are few other compares against compact that are still compared to struct_v and not struct_v.v